packages/calcite-components/src/components/table/table.tsx

```diff
@@ -0,0 +1,510 @@
+    });
+  };
+
+  private updateSelectedItems = (emit?: boolean): void => {
+    const selectedItems = this.bodyRows?.filter((el) => el.selected);
+    this.selectedItems = selectedItems;
+    this.selectedCount = selectedItems?.length;
+    this.allRows?.forEach((row) => {
+      row.selectedRowCount = this.selectedCount;
+      row.selectedRowCountLocalized = this.localizeNumber(this.selectedCount);
+    });
+    if (emit) {
+      this.calciteTableSelect.emit();
+    }
+  };
+
+  private handleDeselectAllRows = (): void => {
+    this.bodyRows?.forEach((row) => {
+      row.selected = false;
+    });
+    this.updateSelectedItems(true);
+  };
+
+  private setSelectedItems = (elToMatch?: HTMLCalciteTableRowElement): void => {
+    this.bodyRows?.forEach((el) => {
+      if (elToMatch?.rowType === "head") {
+        el.selected = this.selectedCount !== this.bodyRows?.length;
+      } else {
+        el.selected =
+          elToMatch === el ? !el.selected : this.selectionMode === "multiple" ? el.selected : false;
+      }
+    });
+    this.updateSelectedItems(true);
+  };
+
+  private localizeNumber = (value: number | string): string => {
+    numberStringFormatter.numberFormatOptions = {
+      locale: this.effectiveLocale,
+      numberingSystem: this.numberingSystem,
+      useGrouping: this.groupSeparator,
+    };
+
+    return numberStringFormatter.localize(value.toString());
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  renderSelectionArea(): VNode {
+    const outOfViewCount = this.selectedItems?.filter((el) => el.hidden)?.length;
+    const localizedOutOfView = this.localizeNumber(outOfViewCount?.toString());
+    const localizedSelectedCount = this.localizeNumber(this.selectedCount?.toString());
+    const selectionText = `${localizedSelectedCount} ${this.messages.selected}`;
+    const outOfView = `${localizedOutOfView} ${this.messages.hiddenSelected}`;
+
+    return (
+      <div class={CSS.selectionArea}>
+        <calcite-chip
+          kind={this.selectedCount > 0 ? "brand" : "neutral"}
+          scale={this.scale}
+          value={selectionText}
+        >
+          {selectionText}
+        </calcite-chip>
+        {outOfViewCount > 0 && (
+          <calcite-chip icon="hide-empty" scale={this.scale} title={outOfView} value={outOfView}>
+            {localizedOutOfView}
+          </calcite-chip>
+        )}
+        {this.selectedCount > 0 && (
+          <calcite-button
+            icon-start="x"
+            kind="neutral"
+            onClick={this.handleDeselectAllRows}
+            round
+            scale={this.scale}
+            title={`${this.messages.clear} ${selectionText} ${this.messages.row}`}
+          >
+            {this.messages.clear}
+          </calcite-button>
+        )}
+        <div class={CSS.selectionActions}>
```

Comment:
Yeah, we can document to use one slotted Action Bar with multiple actions inside vs. slotting multiple individual actions... But I don't want to put all slotted content in an Action Bar - it could be a button, link, plain text, etc.